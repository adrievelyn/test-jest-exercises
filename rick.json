{
    "id": 1,
    "name": "Rick Sanchez",
    "status": "Alive",
    "specie": "HUman",
    "gender": "Male"
}