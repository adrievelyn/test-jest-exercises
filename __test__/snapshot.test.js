import { getCharacter } from '../snapshot';
import rick from '../rick.json';



describe('Es hora de las instantáneas', () => {
    test('Snapshot', () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    });

    test('Siempre fallará la instantánea', () => {
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 20),
        }
        expect(user).toMatchSnapshot();
    });

    test('Tenemos una excepción', () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            name: 'Adriana Sánchez'
        }
        expect(user).toMatchSnapshot({
            id: expect.any(Number)
        });
    });
});