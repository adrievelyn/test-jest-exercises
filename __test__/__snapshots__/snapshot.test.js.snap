// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantáneas Siempre fallará la instantánea 1`] = `
Object {
  "createAt": 2021-07-25T02:45:19.017Z,
  "id": 16,
}
`;

exports[`Es hora de las instantáneas Snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": undefined,
  "status": "Alive",
}
`;

exports[`Es hora de las instantáneas Tenemos una excepción 1`] = `
Object {
  "id": Any<Number>,
  "name": "Adriana Sánchez",
}
`;
